app/src/main/cpp/skyline/applet/swkbd/software_keyboard_applet.cpp

```diff
@@ -0,0 +1,165 @@
+            Utf8Utf16Converter::state_type convert_state;
+            const char16_t *from_next;
+            char8_t *to_next;
+            Utf8Utf16Converter().out(convert_state, text.data(), text.data() + text.size(), from_next, reinterpret_cast<char8_t *>(chars.data()), reinterpret_cast<char8_t *>(chars.data() + chars.size()), to_next);
+            if (to_next < reinterpret_cast<char8_t *>(chars.data()) + chars.size())
+                *to_next = u8'\0';
+        } else {
+            std::memcpy(chars.data(), text.data(), std::min(text.size() * sizeof(char16_t), chars.size()));
+            if (text.size() * sizeof(char16_t) < chars.size())
+                *(reinterpret_cast<char16_t *>(chars.data()) + text.size()) = U'\0';
+        }
+    }
+
+    SoftwareKeyboardApplet::SoftwareKeyboardApplet(
+        const DeviceState &state,
+        service::ServiceManager &manager,
+        std::shared_ptr<kernel::type::KEvent> onAppletStateChanged,
+        std::shared_ptr<kernel::type::KEvent> onNormalDataPushFromApplet,
+        std::shared_ptr<kernel::type::KEvent> onInteractiveDataPushFromApplet,
+        service::applet::LibraryAppletMode appletMode)
+        : IApplet(state,
+                  manager,
+                  std::move(onAppletStateChanged),
+                  std::move(onNormalDataPushFromApplet),
+                  std::move(onInteractiveDataPushFromApplet),
+                  appletMode) {
+        if (appletMode != service::applet::LibraryAppletMode::AllForeground)
+            throw exception("Inline Software Keyboard not implemeted");
+    }
+
+    Result SoftwareKeyboardApplet::Start() {
+        std::scoped_lock lock{inputDataMutex};
+        auto commonArgs{normalInputData.front()->GetSpan().as<service::applet::CommonArguments>()};
+        normalInputData.pop();
+
+        auto configSpan = normalInputData.front()->GetSpan();
+        normalInputData.pop();
+        config = [&] {
+            if (commonArgs.apiVersion < 0x30007)
+                return KeyboardConfigVB{configSpan.as<KeyboardConfigV0>()};
+            else if (commonArgs.apiVersion < 0x6000B)
+                return KeyboardConfigVB{configSpan.as<KeyboardConfigV7>()};
+            else
+                return configSpan.as<KeyboardConfigVB>();
+        }();
+        Logger::Info("Swkbd Config:");
+        Logger::Info(" KeyboardMode: {}", static_cast<u32>(config.commonConfig.keyboardMode));
+        Logger::Info(" InvalidCharFlags: {:b}", *reinterpret_cast<u32*>(&config.commonConfig.invalidCharFlags));
+        Logger::Info(" TextMaxLength: {}", config.commonConfig.textMaxLength);
+        Logger::Info(" TextMinLength: {}", config.commonConfig.textMinLength);
+        Logger::Info(" PasswordMode: {}", static_cast<u32>(config.commonConfig.passwordMode));
+        Logger::Info(" InputFormMode: {}", static_cast<u32>(config.commonConfig.inputFormMode));
+        Logger::Info(" IsUseNewLine: {}", static_cast<u32>(config.commonConfig.isUseNewLine));
```

Comment:
Not sure if this is fit for info level, probably debug would be better. Also since we're going with an async logger in the future, to avoid interleaving this with other stuff change this to
```suggestion
Logger::Info("Swkbd Config:\n* KeyboardMode: {}\n* InvalidCharFlags: 0b{:b}\n* TextMaxLength: {}\n* TextMinLength: {}\n* PasswordMode: {}\n* InputFormMode: {}\n* IsUseNewLine: {}",
    static_cast<u32>(config.commonConfig.keyboardMode),
    *reinterpret_cast<u32*>(&config.commonConfig.invalidCharFlags),
    config.commonConfig.textMaxLength,
    config.commonConfig.textMinLength,
    static_cast<u32>(config.commonConfig.passwordMode),
    static_cast<u32>(config.commonConfig.inputFormMode),
    static_cast<u32>(config.commonConfig.isUseNewLine)
);
```